product/internal/grpc/impl/product_service.go

```diff
@@ -22,7 +24,7 @@ func NewProductService(r repo.ProductRepository) *ProductService {
 // AddProduct method will create a new product.
 func (p *ProductService) AddProduct(ctx context.Context, in *domain.AddProductRequest) (*domain.AddProductResponse, error) {
 	// persist the product
-	err := p.store.CreateProduct(in)
+	_, err := p.store.CreateProduct(in.Product)
 	if err != nil {
```

Comment:
`if _, err := p.store.CreateProduct(in.Product); err != nil {`

Reply:
nice catch!

Reply:
but I have completed the implementation